### PR TITLE
Date Format Functionality Added

### LINKED ISSUE #7650: Add date formatting functionality
### Description

To avoid repetition, either add a date formatting helper function or add a date library such as date fns.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Vishrut19**
## Fixes Issue

Closes #7650 

## Changes proposed
Added Date Format Functionality using formatDate Helper Function.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2023-07-01 at 8 53 48 PM
## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7978"></a>

### CODE REVIEW

**Comment by Vishrut19** [comment]
I know there will be some tailwind css formatting which is not expecting but I don't know how to fix it as my prettier is taking that config directly and is not taking project's config

**Comment by eddiejaoude** [comment]
> I know there will be some tailwind css formatting which is not expecting but I don't know how to fix it as my prettier is taking that config directly and is not taking project's config

Maybe disable the plugin for now and remove the extra changes, it makes it more difficult to review a PR that has unrelated changes

**Comment by Vishrut19** [comment]
> > I know there will be some tailwind css formatting which is not expecting but I don't know how to fix it as my prettier is taking that config directly and is not taking project's config
> 
> Maybe disable the plugin for now and remove the extra changes, it makes it more difficult to review a PR that has unrelated changes

okay I will do it right away and Please review the PR

**Comment by Vishrut19** [comment]
> > I know there will be some tailwind css formatting which is not expecting but I don't know how to fix it as my prettier is taking that config directly and is not taking project's config
> 
> Maybe disable the plugin for now and remove the extra changes, it makes it more difficult to review a PR that has unrelated changes

Changes done. Please review

**Comment by eddiejaoude** [comment]
@Vishrut19 please stop merging in main into your branch on every change in main, it sends me a notification every time for your PR

**Comment by Vishrut19** [comment]
> @Vishrut19 please stop merging in main into your branch on every change in main, it sends me a notification every time for your PR

okay so please tell me how to update the branch, like should I create a new branch for this feature and raise a new PR?

Just confirming please let me know.

**Comment by eddiejaoude** [comment]
It is good to keep your branch up to date, especially if you are still making changes to your branch, but not on every commit to `main` - but as you are not making further changes it should be fine, if a conflict arises then update the branch then and resolve conflicts

**Comment by Vishrut19** [comment]
> It is good to keep your branch up to date, especially if you are still making changes to your branch, but not on every commit to `main` - but as you are not making further changes it should be fine, if a conflict arises then update the branch then and resolve conflicts

okay got it. Did you check the PR? Is it fine?

**Comment by eddiejaoude** [comment]
> > It is good to keep your branch up to date, especially if you are still making changes to your branch, but not on every commit to `main` - but as you are not making further changes it should be fine, if a conflict arises then update the branch then and resolve conflicts
> 
> okay got it. Did you check the PR? Is it fine?

I am working through PRs at the moment

**Comment by Vishrut19** [comment]
> > > It is good to keep your branch up to date, especially if you are still making changes to your branch, but not on every commit to `main` - but as you are not making further changes it should be fine, if a conflict arises then update the branch then and resolve conflicts
> > 
> > 
> > okay got it. Did you check the PR? Is it fine?
> 
> I am working through PRs at the moment

okay got it. If I can be of any help then please let me know

**Comment by eddiejaoude** [comment]
> okay got it. If I can be of any help then please let me know

thank you 👍 (testing PRs is helpful, check them for bugs etc if you have time)

**Comment by Vishrut19** [comment]
> > okay got it. If I can be of any help then please let me know
> 
> thank you 👍 (testing PRs is helpful, check them for bugs etc if you have time)

Sure! Is there any code to test them or we can do it manually?

**Comment by eddiejaoude** [comment]
> > > okay got it. If I can be of any help then please let me know
> > 
> > 
> > thank you 👍 (testing PRs is helpful, check them for bugs etc if you have time)
> 
> Sure! Is there any code to test them or we can do it manually?

Testing their changes, making sure it does what it is meant to do and doesn't break anything else - also the automated tests need more coverage, this requires coding them. We can chat more in Discord so we don't create GitHub notifications for everyone

**Comment by Vishrut19** [comment]
> > > > okay got it. If I can be of any help then please let me know
> > > 
> > > 
> > > thank you 👍 (testing PRs is helpful, check them for bugs etc if you have time)
> > 
> > 
> > Sure! Is there any code to test them or we can do it manually?
> 
> Testing their changes, making sure it does what it is meant to do and doesn't break anything else - also the automated tests need more coverage, this requires coding them. We can chat more in Discord so we don't create GitHub notifications for everyone

Sure, I would love to understand more about it and help you and LinkFree as much as possible.

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you 👍 I think it is used in other places in the project also, please check the following: (I searched the codebase)

- `pages/account/statistics.js`
- `pages/api/events.js`

**Comment by Vishrut19** [comment]
> Thank you 👍 I think it is used in other places in the project also, please check the following: (I searched the codebase)
> 
> * `pages/account/statistics.js`
> * `pages/api/events.js`

Changes Done Eddie

File: components/event/EventCard.js [hunk 1]
```diff
@@ -154,4 +142,4 @@ export default function EventCard({ manage, event, usernames }) {
       {manage ? edit(event) : item(event)}
     </li>
   );
-}
+}
```
Comment (eddiejaoude) [inline components/event/EventCard.js:145]: Blank ending line required as marked by GitHub with the red error

```suggestion
}

```

File: pages/api/profiles/[username]/index.js [hunk 2]
```diff
@@ -101,22 +101,11 @@ export async function getUserApi(req, res, username) {
...
-          "en-GB",
-          dateTimeStyle
-        ).format(start);
-        cleanEvent.date.endFmt = new Intl.DateTimeFormat(
-          "en-GB",
-          dateTimeStyle
-        ).format(end);
-
+        cleanEvent.date.startFmt = formatDate(start);
+        cleanEvent.date.endFmt = formatDate(end);
```
Comment (eddiejaoude) [inline pages/api/profiles/[username]/index.js:108]: This was `en-GB` before but now will default to `en-US`? 🤔

File: services/utils/formatDate.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export function formatDate(date, locale = "en-US") {
```
Comment (eddiejaoude) [inline services/utils/formatDate.js:1]: It looks like we were using `en-GB` why switch to `en-US`?

File: services/utils/formatDate.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  let options = {
+    weekday: "long",
+    day: "numeric",
+    month: "long",
+    year: "numeric",
+    hour: "2-digit",
+    minute: "2-digit",
+    second: "2-digit",
+    timeZoneName: "short",
+  };
```
Comment (eddiejaoude) [inline services/utils/formatDate.js:11]: This is not the options used previous, why the change?

File: services/utils/formatDate.js (hunk 3 above)
Comment (Vishrut19) [reply services/utils/formatDate.js:1]: It was a mistake which I forgot to change.

Comment (Vishrut19) [reply services/utils/formatDate.js:1]: Will do it right away

File: services/utils/formatDate.js (hunk 4 above)
Comment (Vishrut19) [reply services/utils/formatDate.js:11]: 6cf7ec37-41ec-4784-a981-20c13dcdd613

I have added `hour`,`minutes` and `seconds` configs because I though that it would me more precise and also it will give uniform time.

The time would be represented like this -
```
a time of 9:5:3 would be output as 09:05:03:

Hour: 9 becomes 09
Minute: 5 becomes 05
Second: 3 becomes 03
```

Comment (Vishrut19) [reply services/utils/formatDate.js:11]: If this is fine, or should I change it?

File: pages/api/profiles/[username]/index.js (hunk 2 above)
Comment (Vishrut19) [reply pages/api/profiles/[username]/index.js:108]: Yes, I have changed it to be en-GB now

File: services/utils/formatDate.js (hunk 4 above)
Comment (eddiejaoude) [reply services/utils/formatDate.js:11]: According to the official docs that type of time is already achieved with the original config 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#examples

Comment (Vishrut19) [reply services/utils/formatDate.js:11]: okay then I will remove `seconds` thing then

**Comment by Vishrut19** [comment]
Original Config:
Screenshot 2023-07-13 at 12 39 05 PM

Without time config: 
305375b6-13c7-4681-8a21-0c59fd5beb2f

With time config:
Screenshot 2023-07-13 at 12 39 51 PM

Hope this gives you more context @eddiejaoude!

**Comment by eddiejaoude** [comment]
This is how it is now live on production at the moment, the full time is shown, so please use the same config for the moment

Screenshot 2023-07-13 at 08 16 23

**Comment by Vishrut19** [comment]
> This is how it is now live on production at the moment, the full time is shown, so please use the same config for the moment
> 
> Screenshot 2023-07-13 at 08 16 23

okay! So GMT +timezone like for example.
image
(+05:30) Part is not needed?

**Comment by Vishrut19** [comment]
> This is how it is now live on production at the moment, the full time is shown, so please use the same config for the moment
> 
> Screenshot 2023-07-13 at 08 16 23

I am unable to understand what is the change as my output and the output in the production of LinkFree deployed app is the same i think? I am sorry but I am unable to figure out the difference

**Comment by eddiejaoude** [comment]
I think it is best to keep your PR simple and focus, keep the code and config the same, just abstract it out and make it reusable as that was the focus of the issue - any other changes can come in a future PR

**Comment by Vishrut19** [comment]
okay so should I close this PR and also, make use of the FormatDate method of the original component? Like Keeping the same config but just abstracting it?

File: services/utils/formatDate.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export function formatDate(date, locale = "en-GB") {
+  const options = {
+    dateStyle: "full",
```
Comment (eddiejaoude) [inline services/utils/formatDate.js:3]: Not all usage is `full`, for example here https://github.com/EddieHubCommunity/LinkFree/pull/7978/files#diff-bd5cb53f7f62529ac08cd24699132a88e8567b73d93b5289764cfa29442b5720L99 it is `short`

Comment (Vishrut19) [reply services/utils/formatDate.js:3]: okay! But I am confused on this part such that it can be both long and short?

Comment (Vishrut19) [reply services/utils/formatDate.js:3]: Also, I cannot test this how my code is working on statistic page?

Comment (eddiejaoude) [reply services/utils/formatDate.js:3]: Why can't you? You can setup GitHub OAuth for your dev environment https://linkfree.io/docs/environments/environment-variables

File: services/utils/formatDate.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export function formatDate(date, locale = "en-GB") {
```
Comment (eddiejaoude) [inline services/utils/formatDate.js:1]: The defaults we discussed can be added here

Comment (Vishrut19) [reply services/utils/formatDate.js:1]: Sure I will add it, sorry I am a bit busy I will end it asap.

Comment (eddiejaoude) [reply services/utils/formatDate.js:1]: No problem, when you get time - I only asked because you kept chasing me to merge your PR 😂

**Comment by SaraJaoude** [comment]
Thank you for your contribution but as there has been no activity for two weeks I will close this Pull Request.

**Comment by Vishrut19** [comment]
Hey @SaraJaoude I have asked some doubt related to this Issue on discord. Waiting for the reply. So, can you reopen this issue and solve my doubt?
